### PR TITLE
Use smaller N to speed up TestForeach

### PR DESCRIPTION
**Pull request by samestep**
Between September 25 and September 27, approximately half an hour was added to the running time of `pytorch_linux_xenial_cuda10_2_cudnn7_py3_gcc7_test`. Judging from the CircleCI data, it looks like the majority of the new time was added by the following PRs:

- #44550
- #45298

I'm not sure what to do about #44550, but it looks like #45298 increased the `N` for `TestForeach` from just 20 to include both 30 and 300. This PR would remove the 300, decreasing the test time by a couple orders of magnitude (at least when running it on my devserver), from over ten minutes to just a few seconds.

### CODE REVIEW

**Comment by walterddr** [comment]
@izdeby could you please take a look? #45298 introduced two test cases `N = [30, 300]` and I was wondering if there's particular reason for choosing these 2 number and if possible can we reduce the size of the test case?

**Comment by ngimel** [comment]
300 was added there for a reason, to catch a particular failure mode that took quite a long time to debug.

File: test/test_foreach.py [hunk 1]
```diff
@@ -3,6 +3,8 @@
 from torch.testing._internal.common_utils import TestCase, run_tests
 from torch.testing._internal.common_device_type import instantiate_device_type_tests, dtypes, skipCUDAIfRocm
 
+N_values = [30]
```
Comment (malfet) [inline test/test_foreach.py:6]: @izdeby do you know how `[30, 300]` values have been picked originally?
```suggestion
N_values = [20, 50]
```

**Review by malfet — APPROVED** [review]

**Review by ngimel — CHANGES_REQUESTED** [review]
We cannot just remove large number, it is their to test for a failure.

**Comment by samestep** [comment]
@ngimel could you elaborate on what the particular failure mode is? e.g. is there an existing discussion to link to from here?

**Comment by malfet** [comment]
@ngimel, see my change suggestion, i.e. would it be OK to keep large number only for slow tests config?

**Comment by izdeby** [comment]
@samestep, @malfet we need 30 and 300 as these values test two cases - when all tensors can fit into 1 batch and when they can't. Not sure if there is a public discussion on this topic as it was discussed separately with @mcarilli on Slack. 

I think pushing 300 to slow test is ok if we always run those on CI

File: test/test_foreach.py (hunk 1 above)
Comment (izdeby) [reply test/test_foreach.py:6]: yes. these values represent the sizes which fit into 1 batch for multi tensor apply and dont. we definitely want both values.

**Review by walterddr — APPROVED** [review]

**Review by ngimel — APPROVED** [review]

**Pull request closed without merging**